### PR TITLE
Suggest file names for source command.

### PR DESCRIPTION
**Pull request by j-bennet**
## Description

This would add file name completion to `source` command.

File name completion is not "smart", just a simple prefix match.

Related issue: https://github.com/dbcli/mycli/issues/500.

## Checklist

- [X] I've added this contribution to the `changelog.md`.
- [x] I've added my name to the `AUTHORS` file (or it's already there).

### CODE REVIEW

**Comment by tsroten** [comment]
I'm excited to try this out!!! I'm traveling until Tuesday, so I probably won't be able to review it until then. But, I'm sure someone else will give it a shot :-)

**Comment by j-bennet** [comment]
@tsroten @meeuw I would appreciate it if you have time to poke at it (and maybe discover some bugs).

File: mycli/packages/completion_engine.py [hunk 1]
```diff
@@ -289,6 +289,8 @@ def suggest_based_on_last_token(token, text_before_cursor, full_text, identifier
                 prev_keyword, text_before_cursor, full_text, identifier)
         else:
             return []
+    elif text_before_cursor.lower().startswith('source'):
```
Comment (meeuw) [inline mycli/packages/completion_engine.py]: I think we also should support the `\.` command

**Comment by meeuw** [comment]
The completion doesn't work when combined with another query. For example:
```
mysql root@172.17.0.2:dmtest> truncate table test ; source /tmp/test.sql
```
doesn't give completions

**Comment by meeuw** [comment]
This is superb, I hit tab so many times with the \. command :smile:

**Comment by j-bennet** [comment]
@meeuw Weird, it actually works for me combined with another query. I added a test [here](https://github.com/dbcli/mycli/blob/608a30721a377be8470178205d72aaafdb2520d4/test/test_completion_engine.py#L500-L508) to make sure.

**Comment by meeuw** [comment]
Hmm now it works; maybe I wasn't testing on the right branch, sorry.

Could you please also add the `\.` alias? I think the `source` command should be added to `suggest_special` instead of `suggest_based_on_last_token` (not sure, haven't tried that).

**Comment by j-bennet** [comment]
@meeuw Yes, I'm working on `./` alias. Right now, `mycli` won't even execute it:

```
mysql root@localhost:(none)> ./zzz.sql
(1064, u"You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near './zzz.sql' at line 1")
```

so it requires more than just the autocompletion part.

**Comment by tsroten** [comment]
@j-bennet Thanks for working on this!! I think it's `\.` followed by a space.

**Comment by j-bennet** [comment]
@tsroten Ah ok. Thanks for the tip! It's the same with the space though.

```
mysql root@localhost:(none)> . zzz.sql
(1064, u"You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '. zzz.sql' at line 1")
mysql root@localhost:(none)> . ./zzz.sql
(1064, u"You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '. ./zzz.sql' at line 1")
```

**Comment by tsroten** [comment]
@j-bennet Haha, sorry I wasn't clear, I mean like this:
```
localhost:em> \. /tmp/test.sql

```

**Comment by j-bennet** [comment]
@tsroten Oh, I thought it was just a dot. That's what this says http://tldp.org/HOWTO/Bash-Prompt-HOWTO/x237.html. Is it incorrect?

**Comment by j-bennet** [comment]
...but in `mycli` it's a special command `\.`, I see what you mean. Ok.

**Comment by meeuw** [comment]
@j-bennet exactly, `source` / `.` are bash commands and I think these inspired the `mysql` authors (the official mysql client has `source` and `\.` as well.)

**Comment by j-bennet** [comment]
File name is now completed for `\.` command also. The handling of both `\.` and `source` is moved to `suggest_special`.

**Comment by meeuw** [comment]
Thanks a lot @j-bennet this is very useful! I'm going this use this a lot.

**Pull request merged**